lib/bolt/catalog.rb

```diff
@@ -28,30 +32,15 @@ def with_puppet_settings(hiera_config)
       end
     end
 
-    def setup_node(node, trusted)
-      facts = Puppet.lookup(:pal_facts)
-      node_facts = Puppet::Node::Facts.new(Puppet[:node_name_value], facts)
-      node.fact_merge(node_facts)
-
-      node.parameters = node.parameters.merge(Puppet.lookup(:pal_variables))
-      node.trusted_data = trusted
-    end
-
-    def compile_node(node)
-      # Add boltlib to system loaders so modules can use its functions without an
-      # explicit dependency.
-      node.environment.loaders = Bolt::Catalog::BoltLoaders.new(node.environment)
-
-      compiler = Puppet::Parser::BoltCompiler.new(node)
-      compiler.compile(&:to_resource)
-    end
-
     def generate_ast(code)
       with_puppet_settings do
-        Puppet::Pal.in_tmp_environment("bolt_parse") do |_pal|
-          node = Puppet.lookup(:pal_current_node)
-          compiler = Puppet::Parser::BoltCompiler.new(node)
-          compiler.dump_ast(compiler.parse_string(code))
+        Puppet::Pal.in_tmp_environment("bolt_parse") do |pal|
+          pal.with_catalog_compiler do |compiler|
```

Comment:
I think you can maybe set trusted here with something like

`Puppet.lookup(:pal_current_node).trusted_data = trusted_data`